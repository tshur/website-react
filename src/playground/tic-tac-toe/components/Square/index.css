:local .square {
  background: #eee;
  border: 1px solid #777;
  float: left;
  padding-top: 2px;
  margin: 0;
  height: 60px;
  width: 60px;
  line-height: 60px;
  text-align: center;
  font-size: 48px;
  overflow: hidden;
  word-wrap: none;
}

:local .square:focus {
  outline: 0;
}