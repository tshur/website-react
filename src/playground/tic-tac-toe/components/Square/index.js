import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const Square = ({ value, onClick, winningSquare }) =>
  <button
    className={`${styles.square} ${winningSquare ? styles.highlight : ''}`}
    onClick={onClick}
  >
    {value}
  </button>

Square.defaultProps = {
  value: null,
  winningSquare: false
};

Square.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  winningSquare: PropTypes.bool
}

export default Square;