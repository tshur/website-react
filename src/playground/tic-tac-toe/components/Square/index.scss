.square {
  background: #eee;
  border: 0;
  border-right: 3px solid #444;
  float: left;
  padding-top: 4px;
  margin: 0;
  height: 60px;
  width: 60px;
  line-height: 60px;
  text-align: center;
  font-size: 48px;
  overflow: hidden;
  word-wrap: none;

  &:last-child {
    border-right: 0;
  }

  &:focus {
    outline: 0;
  }
}

.highlight {
  background: #dd2222;
}
