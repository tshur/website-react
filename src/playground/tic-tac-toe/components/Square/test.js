import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Square from '.';

describe('Square', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Square />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Square />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
