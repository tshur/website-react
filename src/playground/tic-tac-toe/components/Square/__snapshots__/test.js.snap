// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square has a valid snapshot 1`] = `
<button
  className="undefined "
  onClick={[Function]}
/>
`;
